### DISCUSSION TITLE
Can pocketbase be used to do Server Side Rendering and return HTML?

### DISCUSSION THREAD

**Discussion by baldr9** [pocketbase/pocketbase#2091] [Q&A] [answered]
I like the [htmx.org](https://htmx.org/) model of using server side rendered HTML as the response to an API call.

Is anyone using pocketbase with API responses that return HTML?

Since pocketbase is using Echo for the routes, I saw the documentation suggests using:
* https://pocketbase.io/docs/event-hooks#app-hooks
* https://pocketbase.io/docs/router/

So I would need to create custom routes, use Echo with templates, to return the HTML
```
func htmxTest(c echo.Context) error {
  return c.Render(http.StatusOK, "htmx-test-response", template.HTML("<p>HTMX Test</p>"))`
}
```

**Comment by dschissler** [comment, marked as answer]
To be able to use `Render` on the context you first need to define that function.  Inside that function you'll locate the template resources, compile them and then return the data.  Depending on if you want to have common layouts or common data like if the user is logged in then you might have quite a lot of work to do.

**Comment by efim** [comment]
I was really interested in using Pocketbase for SSR go + templates + htmx , especially having oauth avaialble for signup.
Here are results of my moderately successful experimentation:
repo: http://git.sunshine.industries/efim/go-ssr-pocketbase-oauth-attempt (with Readme)
and deployed site: https://go-ssr-oauth-attempt.sunshine.industries/

I am certain things can be radically improved, since i'm a beginner in go,
but having htmx rendered pages, integrating into authentication, and modifying error responses to show hypertext is not too hard with pocketbase
